### DISCUSSION TITLE
Example implementation for generating subs for audio element

### DISCUSSION THREAD

**Discussion by n-ce** [alexandercerutti/sub37#8] [Q&A] [answered]
### Hi, 
I would request an example implementation for how to use sub37 for getting subtitles for an audio element, specifically for showing subs from ttml files.
I did try to implement one myself but it didn't work out in the end due to the complexity involved.

I have been using the imscjs library for the same so far, but due to lack of TS support, I had to deprecate support for it, so I have been wanting to use your library instead now.
If you were curious, it would be used in a [FOSS project](https://github.com/n-ce/ytify).
### Thank you.

**Comment by alexandercerutti** [comment, marked as answer]
Hi @n-ce, thanks for considering sub37!

First of all, let me say that sub37 feels like to be yet at an early stage of development. In fact, I'm still working on the adapter for TTML subtitles (it is available on the branch `feature/ttml-adapter`), while WebVTT subtitles adapter is already available. Implementing TTML support is indeed revealing a lot of complexity, due its standard. I'm proceeding very cautiously because I know I might encounter a specification that might require me to change everything I did before.

Second (if I understood correctly your query), sub37 does not provide a mechanism to extract or generate subtitles from an audio track. It also does not support (yet? not planned) the reading of a subtitles from videos.

I hope I was able to answer your doubts.

**Comment by n-ce** [comment]
Thank you for your response.
Did you mean it cannot process ttml files and show it in the DOM currently? That's what ImscJS does it takes an input of the ttml file and the audio/video element and converts ttml to webvtt and then displays the subs in the view.

**Reply by alexandercerutti** [reply]
Exactly. The purpose of this library is to create something completely different, that does not use the WebVTT browser API, as they differ in implementation and support across browsers. The idea is to have modular support for different file formats and have dedicated parsers that convert to an internal common and known format, defined by `@sub37/server`.

Also, a great way to learn algorithms and so on. 😄
